qttypes/src/gui/qcolor.rs

```diff
@@ -0,0 +1,871 @@
+pub use crate::core::{qreal, QString};
+
+#[cfg(not(no_qt))]
+use cpp::{cpp, cpp_class};
+
+cpp! {{
+    #include <QtGui/QColor>
+    #include <QtCore/QString>
+}}
+
+#[derive(Copy, Clone, Eq, PartialEq, Hash)]
+pub struct QRgb(u32);
+
+impl QRgb {
+    pub fn alpha(&self) -> u8 {
+        ((self.0 >> 24) & 0x000000ff) as u8
+    }
+    pub fn red(&self) -> u8 {
+        ((self.0 >> 16) & 0x000000ff) as u8
+    }
+    pub fn green(&self) -> u8 {
+        ((self.0 >> 8) & 0x000000ff) as u8
+    }
+    pub fn blue(&self) -> u8 {
+        (self.0 & 0x000000ff) as u8
+    }
+}
+
+impl From<u32> for QRgb {
+    fn from(val: u32) -> QRgb {
+        QRgb(val)
+    }
+}
+
+impl Into<u32> for QRgb {
+    fn into(self) -> u32 {
+        self.0
+    }
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Hash)]
+pub struct QRgba64(u64);
+impl QRgba64 {
+    pub fn alpha(&self) -> u16 {
+        ((self.0 >> 48) & 0x0000ffff) as u16
+    }
+    pub fn red(&self) -> u16 {
+        ((self.0 >> 32) & 0x0000ffff) as u16
+    }
+    pub fn green(&self) -> u16 {
+        ((self.0 >> 16) & 0x0000ffff) as u16
+    }
+    pub fn blue(&self) -> u16 {
+        (self.0 & 0x0000ffff) as u16
+    }
+}
+
+impl From<u64> for QRgba64 {
+    fn from(val: u64) -> QRgba64 {
+        QRgba64(val)
+    }
+}
+
+impl Into<u64> for QRgba64 {
+    fn into(self) -> u64 {
+        self.0
+    }
+}
+
+/// Bindings for [`QColor::NameFormat`][class] enum class.
+///
+/// [class]: https://doc.qt.io/qt-5/qcolor.html#NameFormat-enum
+#[repr(u32)]
+#[derive(Clone, Copy, PartialEq, Debug)]
+#[allow(non_camel_case_types)]
+pub enum NameFormat {
+    // #RRGGBB A "#" character followed by three two-digit hexadecimal numbers (i.e. #RRGGBB).
+    HexRgb = 0,
+    //#AARRGGBB A "#" character followed by four two-digit hexadecimal numbers (i.e. #AARRGGBB).
+    HexArgb = 1,
+}
+
+/// Bindings for [`QColor::Spec`][class] enum class.
+///
+/// [class]: https://doc.qt.io/qt-5/qcolor.html#Spec-enum
+#[repr(u32)]
+#[derive(Clone, Copy, PartialEq, Debug)]
+#[allow(non_camel_case_types)]
+pub enum Spec {
```

Comment:
This should be named QColorSpec,  and NameFormat  QColorNameFormat, because it otherwise is not descriptive enough as it is outside of a namespace.